tests/lib-tests.nix

```
@@ -374,6 +374,28 @@ let
         ];
       };
     };
+
+    testEscapeStringForLua = {
+      expr = {
+        simple = helpers.utils.escapeStringForLua "simple";
+        depth-one = helpers.utils.escapeStringForLua "]]";
+      };
+      expected = {
+        simple = "[[simple]]";
+        depth-one = "[=[]]]=]";
+      };
+    };
+
+    testEscapeStringForVimscript = {
+      expr = {
+        simple = helpers.utils.escapeStringForVimscript "simple";
+        depth-one = helpers.utils.escapeStringForVimscript "EOF";
+      };
```

```suggestion
      expr = lib.mapAttrs (_: helpers.utils.escapeStringForVimscript) {
        simple = "simple";
        depth-one = "EOF";
        depth-two = "EOF EOFF";
      };
```

We can use `mapAttrs` to generalize the test-case.